### PR TITLE
Label styles + inline label prop

### PR DESCRIPTION
**Pull request by denbalogh**

### CODE REVIEW

File: storybooknative/src/CenterView/index.js [hunk 1]
```diff
@@ -5,17 +5,18 @@ import { View } from "react-native";
...
   justifyContent: "center",
-  alignItems: "center",
+  alignItems: props.align || "center",
   backgroundColor: "#f5fcff",
-});
+}));
 
 type Props = {|
   children: React.ChildrenArray<React.Node> | null,
+  align: "flex-start" | "center" | "flex-end" | "stretch",
```
Comment (tbergq) [inline storybooknative/src/CenterView/index.js]: Props should be marked readonly `+align: "flex-start" | "center" | "flex-end" | "stretch"`

Also, maybe we should extract this align-enum somewhere and export it. It could be used by other components in the future also?

File: storybooknative/src/FormLabel/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+StyledAsterisk.defaultProps = {
+  theme: defaultTokens,
+};
+
+const StyledLabel = styled(Text, props => {
+  const { theme, disabled, filled } = props;
+  return {
+    fontSize: theme.orbit.fontSizeFormLabel,
+    color: !filled || disabled ? theme.orbit.colorFormLabel : theme.orbit.colorFormLabelFilled,
+    // lineHeight: theme.orbit.lineHeightText, #TODO remove comment before merge
```
Comment (tbergq) [inline storybooknative/src/FormLabel/index.js]: why is this commented?

File: storybooknative/src/FormLabel/index.js.flow [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// @flow
+
+export type Props = {
+  children: React$Element,
+  filled?: boolean,
+  disabled?: boolean,
+  required?: boolean,
```
Comment (tbergq) [inline storybooknative/src/FormLabel/index.js.flow]: read-only

File: storybooknative/src/InputField/index.js [hunk 4]
```diff
@@ -70,8 +91,8 @@ const StyledTextInput = styled(TextInput, props => {
     color: disabled ? theme.orbit.colorTextInputDisabled : theme.orbit.colorTextInput,
     fontSize: getToken(TOKENS.fontSizeInput, props),
     width: "100%",
-    height: "100%",
-    lineHeight: getToken(TOKENS.heightInput, props),
+    height: getToken(TOKENS.heightInput, props),
+    // lineHeight: getToken(TOKENS.heightInput, props), #TODO remove comment before merge
     zIndex: 2,
```
Comment (tbergq) [inline storybooknative/src/InputField/index.js]: Don't we have some zIndex values in orbit-tokens?

File: storybooknative/src/InputField/index.js [hunk 5]
```diff
@@ -94,14 +115,23 @@ const InputField = (props: Props) => {
...
+  const isFilled = !!props.value;
   return (
     <React.Fragment>
-      {label !== undefined && label}
+      {label && !inlineLabel && <FormLabel label={label} isFilled={isFilled} required={required} />}
       <StyledInputContainer size={size} disabled={disabled} error={error}>
+        {label &&
+          inlineLabel && (
+            <StyledInlineLabel size={size} filled={isFilled} disabled={disabled}>
+              {required && <StyledAsterisk filled={isFilled}>* </StyledAsterisk>}
```
Comment (tbergq) [inline storybooknative/src/InputField/index.js]: Maybe <StyledAsterisk could be exported with the `*` so we don't have to fill it every time?

```
Const StyledAsteriskWrapper = styled(Text, {...});
export StyledAsterisk = props => <StyledAsteriskWrapper {...props}>* </StyledAsteriskWrapper>
```

File: storybooknative/src/InputField/index.js (hunk 4 above)
Comment (denbalogh) [reply storybooknative/src/InputField/index.js]: We do but not for this case. These values are from the web version.

Comment (tbergq) [reply storybooknative/src/InputField/index.js]: And why does it matter that they are from the web version?

Comment (denbalogh) [reply storybooknative/src/InputField/index.js]: I thought that web and RN should be as consistent as possible.

File: storybooknative/src/FormLabel/index.js.flow [hunk 6]
```diff
@@ -1,8 +1,12 @@
...
-  filled?: boolean,
-  disabled?: boolean,
-  required?: boolean,
+  +children: React$Element,
+  +filled?: boolean,
+  +disabled?: boolean,
+  +required?: boolean,
+};
+
+export type asteriskProps = {
```
Comment (tbergq) [inline storybooknative/src/FormLabel/index.js.flow]: `AsteriskProps`

File: storybooknative/src/InputField/index.js (hunk 4 above)
Comment (tbergq) [reply storybooknative/src/InputField/index.js]: Ok, let me rephrase, we can't we use zIndex from orbit-tokens?

I am not saying this is wrong here, I just don't understand completely why 🙂
